tests/ga/test_extension.py

```diff
@@ -1961,5 +2026,174 @@ def test_ext_handler_with_systemd_cgroup_enabled(self, *args):
         self._assert_no_handler_status(protocol.report_vm_status)
 
 
+@patch('azurelinuxagent.ga.exthandlers.ExtHandlerInstance._capture_process_output', side_effect="Process Successful")
+class TestExtensionUpdateOnFailure(ExtensionTestCase):
+
+    @staticmethod
+    def _get_ext_handler_instance(name, version, continue_on_update_failure=False):
+
+        handler_json = {
+            "installCommand": "sample.py -install",
+            "uninstallCommand": "sample.py -uninstall",
+            "updateCommand": "sample.py -update",
+            "enableCommand": "sample.py -enable",
+            "disableCommand": "sample.py -disable",
+            "rebootAfterInstall": False,
+            "reportHeartbeat": False,
+            "continueOnUpdateFailure": continue_on_update_failure
+        }
+
+        ext_handler_properties = ExtHandlerProperties()
+        ext_handler_properties.version = version
+        ext_handler = ExtHandler(name=name)
+        ext_handler.properties = ext_handler_properties
+        ext_handler_i = ExtHandlerInstance(ext_handler=ext_handler, protocol=None)
+        ext_handler_i.load_manifest = MagicMock(return_value=HandlerManifest({'handlerManifest': handler_json}))
+        fileutil.mkdir(ext_handler_i.get_base_dir())
```

Comment:
This should be moved/mocked to a temporary folder which gets deleted on test cleanup.

Reply:
The base director `ext_handler_i.get_base_dir()` is already set to a /tmp/ folder so no additional checks are needed.
Eg - `'/tmp/TestExtensionUpdateOnFailure_fk9fsa6r/foo-1.0.0'`